wgsl/extract-grammar.py

```diff
@@ -0,0 +1,741 @@
+#!/usr/bin/env python3
```

Comment:
Is this tied specifically to WebGPU specs in any way? It sounds to me that it could be useful for other specs as well, in which case you'd be better off hosting it on your own project, and we could depend on it as a sub-module here.

Reply:
@kvark I think our expression of grammar in the spec will probably change drastically for better reading experience, which I was thinking of bringing up right after we progress this. Taking into consideration that we might turn to developing the Semantic CI, I think the spec and the extractor might be better off if they evolve mutually together than as a distinct module, but I also see the point. extract-idl is a precedent but as you point out that's kinda more tightly coupled to the WebGPU spec!

Reply:
Ok, I just realized that not every w3C spec carries a separate grammar for a separate language.